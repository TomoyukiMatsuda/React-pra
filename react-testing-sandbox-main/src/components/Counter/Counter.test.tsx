import { cleanup, fireEvent, render, screen } from "@testing-library/react"

import { Counter } from "./"

// "Counter"はテストグループ名、"render"と"click:count"はテストケース
describe("Counter", () => {
  // テストケース終了ごとにReactコンポーネントを画面からunmountするためのクリーンナップ用ヘルパー関数
  afterEach(() => {
    cleanup()
  })
  test("render", () => {
    const { asFragment, getByText } = render(<Counter />)
    expect(asFragment()).toMatchSnapshot()
    // 最初のレンダリングでテキストの表示が "Count: 0" であることをテスト
    expect(getByText("Count: 0"))
  })
  test("click:count", () => {
    render(<Counter />)
    // "Increment"のテキストが入ったbuttonを取得
    const button = screen.getByText("Increment")
    // 上記で取得したボタンを5回クリックすると
    for (let i = 0; i < 5; i++) {
      fireEvent.click(button)
    }
    // 画面のテキストには Count: 5 というテキストが表示される
    screen.getByText("Count: 5")
  })
})
