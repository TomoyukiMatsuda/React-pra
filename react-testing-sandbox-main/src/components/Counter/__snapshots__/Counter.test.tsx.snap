// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter render 1`] = `
<DocumentFragment>
  <div>
    Count: 0
  </div>
  <div>
    <button>
      Increment
    </button>
  </div>
</DocumentFragment>
`;
